source/references/glossary.markdown

```diff
@@ -251,148 +358,231 @@ Generally speaking, a parameter is a chunk of information that a class or resour
+* **Local scope,** where you can refer to a variable by its short name inside that scope and the scope's children, but typically cannot access it from other scopes.
 
-### site
+There are a few exceptions, and you might also encounter other, more situational scopes. For detailed information, see the [Scope](/puppet/latest/reference/lang_scope.html) page in the [Puppet language](#puppet-language) reference.
 
-An entire IT ecosystem being managed by Puppet. That is, a site includes all puppet master servers, all agent nodes, and all independent masterless Puppet nodes within an organization.
+> **Note:** Previously, Puppet used dynamic scope, which would search for short-named variables through a long chain of parent scopes. This scope was deprecated in Puppet 2.7, and in Puppet 4 is used only for resource defaults.
 
-### site manifest
+### singleton
 
-The main "point of entry" [manifest](#manifest) used by the puppet master when compiling a catalog. The location of this manifest is set with the `manifest` setting in puppet.conf. Its default value is usually `/etc/puppet/manifests/site.pp` or `/etc/puppetlabs/puppet/manifests/site.pp`.
+A **singleton** is an object in the [Puppet language](#puppet-language), such as a [class](#class), that can only be evaluated once. For example, you can't have more than one distinct class with the same specific [name](#namevar) in a [manifest](#manifest) or [catalog](#catalog), making that class a singleton.
+
+### site
 
-The site manifest usually contains [node definitions](#node-definition). When an [ENC](#external-node-classifier) is being used, the site manifest may be nearly empty, depending on whether the ENC was designed to have complete or partial node information.
+A **site** refers to an entire IT ecosystem that is managed by Puppet. A site includes all Puppet [master](#master) servers, [agent nodes](#agent), and independent masterless Puppet [nodes](#node) within an organization.
 
 ### site module
 
-A common idiom in which one or more [modules](#module) contain [classes](#class) specific to a given Puppet site. These classes usually describe complete configurations for a specific system or a given group of systems. For example, the `site::db_slave` class might describe the entire configuration of a database server, and a new database server could be configured simply by applying that class to it.
+A **site module** is a common Puppet idiom in which one or more [modules](#module) contain [classes](#class) specific to a given Puppet [site](#site). These classes usually describe complete configurations for a specific system or group of systems. For example, the `site::db_slave` class might describe the entire configuration of a database server, and a new database server could be configured simply by applying that class to it.
```

Comment:
mmmmmmmmm this approach was kind of obsoleted by the roles and profiles pattern. ...which we still don't have great docs for.